### PR TITLE
kinder: remove 1.16 logic and workflows

### PR DESCRIPTION
**Pull request by neolit123**
The latest-1.15.txt version label/marker was removed
due to GCS "retention" (not updated for some time)
and these jobs started failing.

- Remove logic from kinder related to the 1.15 version condition
- Remove the 1.15 skew and upgrade workflows

### CODE REVIEW

File: kinder/pkg/cluster/manager/actions/kubeadm-init.go [hunk 1]
```diff
@@ -209,24 +201,10 @@ func kubeadmInitWithPhases(cp1 *status.Node, automaticCopyCerts bool, kustomizeD
 
 	if automaticCopyCerts {
 		uploadCertsArgs := []string{
-			"init", "phase", "upload-certs",
+			"init", "phase", "upload-certs", "--upload-certs",
 			fmt.Sprintf("--config=%s", constants.KubeadmConfigPath),
 			fmt.Sprintf("--v=%d", vLevel),
 		}
-		if automaticCopyCerts {
```
Comment (neolit123) [inline kinder/pkg/cluster/manager/actions/kubeadm-init.go:216]: this seems to be checked already above.
just appending the flag --upload-certs to uploadCertsArgs seems sufficient here.

File: kinder/pkg/cluster/manager/actions/kubeadm-join.go [hunk 2]
```diff
@@ -111,21 +111,13 @@ func joinControlPlanes(c *status.Cluster, usePhases, automaticCopyCerts bool, di
...
 		"join",
 		fmt.Sprintf("--config=%s", constants.KubeadmConfigPath),
 		fmt.Sprintf("--v=%d", vLevel),
 		constants.KubeadmIgnorePreflightErrorsFlag,
 	}
-	if automaticCopyCerts {
-		// if before v1.15, add certificate key flag (for >= 15, certificate key is passed via the config file)
-		if cp.MustKubeadmVersion().LessThan(constants.V1_15) {
-			joinArgs = append(joinArgs,
-				fmt.Sprintf("--certificate-key=%s", constants.CertificateKey),
```
Comment (neolit123) [inline kinder/pkg/cluster/manager/actions/kubeadm-join.go:125]: making our e2e prefer config seems fine.

File: kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go [hunk 3]
```diff
@@ -170,41 +170,20 @@ func kubeadmUpgradeNode(c *status.Cluster, n *status.Node, upgradeVersion *K8sVe
...
-			}
-		}
-
-		// kubeadm upgrade node config
-		if err := n.Command(
-			"kubeadm", "upgrade", "node", "config", "--kubelet-version", fmt.Sprintf("v%s", upgradeVersion), fmt.Sprintf("--v=%d", vLevel),
-		).RunWithEcho(); err != nil {
-			return err
-		}
+	// kubeadm upgrade node
```
Comment (neolit123) [inline kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go:173]: this is copied from the `if n.MustKubeadmVersion().AtLeast(constants.V1_15) {` branch.

File: kinder/pkg/constants/constants.go [hunk 4]
```diff
@@ -107,12 +107,6 @@ const (
 
 // kubernetes releases, used for branching code according to K8s release or kubeadm release version
 var (
-	// V1_13 minor version
-	V1_13 = K8sVersion.MustParseSemantic("v1.13.0-0")
```
Comment (neolit123) [inline kinder/pkg/constants/constants.go:111]: stray, unused version constant.

File: kinder/pkg/kubeadm/config.go [hunk 5]
```diff
@@ -74,10 +73,6 @@ func getKubeadmConfigVersion(kubeadmVersion *K8sVersion.Version) (string, error)
 	// v1alpha2 (that is Kuberntes v1.11.0) is out of support.
 	// v1alpha3 (that is Kuberntes v1.13.0) is out of support.
 
-	if kubeadmVersion.LessThan(constants.V1_15) {
-		return "v1beta1", nil
```
Comment (fabriziopandini) [inline kinder/pkg/kubeadm/config.go:78]: @neolit123 the fact we are removing "v1beta1" from the kubeadm test grid is kind of worrying because it is still used in CAPI. 

We should consider adding a flag for forcing the config version to use and add dedicated tests for this (v1.16 with v1beta1, v1.16 with v1beta1 etc), but this is out of scope of this PR.

In the meantime, let's preserve int the code configTemplateBetaV1

Comment (fabriziopandini) [reply kinder/pkg/kubeadm/config.go:78]: https://github.com/kubernetes/kubeadm/issues/2249

Comment (neolit123) [reply kinder/pkg/kubeadm/config.go:78]: i will commented on #2249

**Comment by neolit123** [comment]
/retitle kinder: remove 1.16 logic and workflows

**Comment by neolit123** [comment]
/retitle kinder: remove 1.15 logic and workflows
